docs/decisions/0001-meilisearch.rst

```diff
@@ -0,0 +1,146 @@
+   restrictive license that prohibits providing "the products to others as a
+   managed service". Consequently, AWS forked the search engine to create
+   OpenSearch and no longer offers Elasticsearch as a service. This is
+   problematic for many Open edX operators that use AWS and prefer to avoid
+   any third-party services.
+2. Elasticsearch is very resource-intensive and often uses more than a gigabyte
+   of memory just for small search use cases.
+3. Elasticsearch has poor support for multi-tenancy, which multiplies the
+   problem of resource usage for organizations with many small Open edX sites.
+4. The existing usage of edx-search/Elasticsearch routes all search requests and
+   result processing through edxapp (the LMS) or other IDAs, increasing the
+   load on those applications.
+
+Meilisearch
+===========
+
+Meilisearch ("MAY-lee search") is a new, promising search engine that offers a
+compelling alternative to Elasticsearch. It is open source, feature rich, and
+very fast and memory efficient (written in Rust, uses orders of magnitude less
+memory than Elasticsearch for small datasets). It has a simple API with an
+official python driver, and has official integrations with the popular
+Instantsearch frontend library from Algolia. It has strong support for
+multi-tenancy, and allows creating restricted API keys that incorporate a user's
+permissions, so that search requests can be made directly from the user to
+Meilisearch, rather than routing them through Django. Initial testing has shown
+it to be much more developer friendly than Elasticsearch/OpenSearch.
+
+At the time of writing, there are only two known concerns with Meilisearch:
+
+1. It doesn't (yet) support High Availability via replication, although this is
+   planned and under development. It does have other features to support high
+   availability, such as very low restart time (in ms).
+2. It doesn't support boolean operators in keyword search ("red AND panda"),
+   though it does of course support boolean operators in filters. This is a
+   product decision aimed at keeping the user experience simple, and is unlikely
+   to change.
+
+Problem with edx-search
+=======================
+
+The edx-search feature is currently utilized only within the edx-platform, and many of its implementations are now deprecated in the new microfrontend environments. Additionally, it lacks the personalized token functionality, making it incompatible with advanced search engines such as Meilisearch. Even if we attempt to extend edx-search to support Meilisearch, it would still follow the same proxy mechanism and would not be able to utilize personalized tokens effectively. To find a detailed list of architectural issues here.
+
+Open edX Search API
+===================
+
+We are proposing a new abstraction package with more generalised architecture inorder to support all available search engines including elasticsearch, meilisearch and algolia.
```

Comment:
If we require drivers to implement `get_user_token()` then it may not support Elasticsearch/Opensearch ? I guess it _could_, by using a fake token and routing requests through the LMS. But it seems like it would be easier to only support Meilisearch and Algolia.

Reply:
It will be return configuration object with we can use in frontend driver to request elasticsearch.